dynamodb-stream-reader/src/test/java/no/unit/nva/dynamodb/DynamoDBStreamHandlerTest.java

```diff
@@ -0,0 +1,128 @@
+package no.unit.nva.dynamodb;
+
+import com.amazonaws.services.lambda.runtime.Context;
+import com.amazonaws.services.lambda.runtime.events.DynamodbEvent;
+import elasticsearch.ElasticSearchRestClient;
+import nva.commons.utils.Environment;
+import nva.commons.utils.IoUtils;
+import nva.commons.utils.JsonUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.http.HttpClient;
+import java.net.http.HttpResponse;
+import java.nio.file.Paths;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+
+@SuppressWarnings("unchecked")
+public class DynamoDBStreamHandlerTest {
+
+    private static final String SAMPLE_MODIFY_EVENT_FILENAME = "DynamoDBStreamModifyEvent.json";
+    private static final String SAMPLE_INSERT_EVENT_FILENAME = "DynamoDBStreamInsertEvent.json";
+    private static final String SAMPLE_REMOVE_EVENT_FILENAME = "DynamoDBStreamRemoveEvent.json";
+    private static final String SAMPLE_UNKNOWN_EVENT_FILENAME = "UnknownDynamoDBEvent.json";
+    public static final String ELASTICSEARCH_ENDPOINT_ADDRESS = "localhost";
+    private static final String ELASTICSEARCH_ENDPOINT_INDEX = "resources";
+    private static final String ELASTICSEARCH_ENDPOINT_API_SCHEME = "http";
+    private static final Object TARGET_SERVICE_URL = "http://localhost/service/";
+
+    private DynamoDBStreamHandler handler;
+    private ElasticSearchRestClient elasticSearchRestClient;
+    private Environment environment;
+    private Context context;
+    private HttpClient httpClient;
+
+    /**
+     * Set up test environment.
+     *
+     * @throws IOException some error occurred
+     * @throws InterruptedException another error occurred
+     */
+    @BeforeEach
+    public void init() throws IOException, InterruptedException {
+        environment = spy(Environment.class);
+        context = mock(Context.class);
+        httpClient = spy(HttpClient.class);
+        doReturn(ELASTICSEARCH_ENDPOINT_ADDRESS).when(environment)
```

Comment:
1. Why spy and not mock?
2. gather all Environment setup in one method called `mockEnvirornment` or something .

Reply:
Mock works just fine...  Environment setup moved to own method